### DISCUSSION TITLE
Periodical Glitches clicking on sound (typical buffer too small sound)

### DISCUSSION THREAD

**Discussion by LALAMUSIC** [ExistentialAudio/BlackHole#100] [General]
Hello, 
I have been testing blackhole and the software works great : ) but I sometime have glitches (typical sample size too small effect) .

My setup is the following :
Soundcard (ROLAND quad capture) in 1-2(Mics)( 128 samples) to DAW-> DAW to Jamkazam(music brodcast soft) using blackhole ->Jamkazam to soundcard Out 1-2 connected to speakers.

The glitches appear periodically every 1 min and 24 seconds approximately. Is this a clock sync issue beetween my soundcard and black hole ? Do you have any idea ?

Thanks for helping

**Comment by devinroth** [comment]
My first inclaination is it’s a syncing issue. Are you using an aggregate device?

**Comment by LALAMUSIC** [comment]
Hi, thank you for your fast answer. 
No, i'm not using  any aggregate device in this setup.

**Comment by devinroth** [comment]
Try using an aggregate with delay compensation enabled in your DAW. I suspect that will solve your problem. What DAW are you using?

**Comment by LALAMUSIC** [comment]
I tried tour idea and it seems to change , now I have the glitch coming after 2 mins.
I don't know where to act on delay compensation and which values to set .
DAW is Live

Also who should be the master clock in the aggregate device ? soundcard or BlackHole ?

Actual routing : Live outputs at out 5/6 of the aggregate and Jamkazam receive at in 7/8 of the aggregate and outputs at out 1/2 of the aggregate to speakers.
![Capture d’écran 2020-04-12 à 20 27 44](https://user-images.githubusercontent.com/63554857/79076783-13dc5b80-7cfd-11ea-8c70-b1baf1a04bf2.png)
![Capture d’écran 2020-04-12 à 20 28 05](https://user-images.githubusercontent.com/63554857/79076785-15a61f00-7cfd-11ea-8974-bc3dae51f817.png)

**Comment by devinroth** [comment]
Keep everything the same but Check delay compensation (Correction de dérive) on BlackHole from Audio Midi Setup. The one in Live does something else.

Try it out and let me know.

**Comment by LALAMUSIC** [comment]
It seems to work well ticking delay compensation on BlackHole from Audio Midi Setup.
Some glitch occur but very seldom and make it very acceptable. 
Does Ticking both delay compensation on Quad Capture AND Black Hole would help more ?

Thank you for your support and for the great BlackHole Tool.

**Comment by devinroth** [comment]
You can try both but I don’t suspect it will improve any more. Unfortunately Aggregates are not the best. You can also try switching it around with BlackHole as the primary clock to see if it improves more. 

I’m trying to work on a solution to sync BlackHole clock with any audio device. It’s a work in progress.

**Comment by devinroth** [comment]
What settings got the best result?

**Comment by LALAMUSIC** [comment]
I sticked with the delay compensation activated on BlackHole and the Soundcard's clock it works well.

I will let you know if I change settings and see any improvements or changes.

Nice to hear that you're developing the sync feature. I will check for release

**Comment by heitorsilva** [comment]
Hey @LALAMUSIC!

I've been testing Blackhole heavily in the last two days, over a Skype call. I'm streaming my DAW audio to the Skype call, and two things that helped with the quality the other end of the call was hearing, besides the delay compensation, was using 512 samples as buffer size in the DAW, and raising the format to 192.000 Hz on Audio MIDI Setup.

Try raising your config from 44.1 Hz to the highest frequency you have available, and see if that will help you out as helped me on the Skype call. Don't forget the sample size for buffering.

Let us know!

Cheers, stay safe...

**Reply by topazcoding** [reply]
I have had the same issues. random glitching (like latency too low) using high sierra, mac Pro 5.1 Motu Ultralite
Same issue in cubase and ableton.

both aggregate io out and only BlackHole out. tried all of the above but still get those random glitches.

I am basically routing audio out of my DAW into Element (stand alone host) and then thru IK Ark etc to achieve system wide corretion.

**Comment by devinroth** [comment]
Are you able to test the 0.2.8 branch? That may solve the issue. I’m interested in knowing.

Devin

**Reply by topazcoding** [reply]
Merry Xmas Devin ;-)

sorry but i'm not sure how to do that. only see 0.27 to download.
